File: src/main/java/com/cfgglobal/test/domain/EmailLog.java
package com.cfgglobal.test.domain;

import com.cfgglobal.test.enums.TaskStatus;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity
@DynamicUpdate
@DynamicInsert
@FieldDefaults(level = AccessLevel.PRIVATE)
public class EmailLog extends BaseEntity implements Serializable {

    String subject;
    @Lob
    @Column(length = 100000)
    @JsonIgnore
    byte[] content;
    String sendTo;
    @Enumerated(value = EnumType.STRING)
    TaskStatus status;
    Integer times;//发送次数
    String msg;
}

File: src/main/kotlin/com/cfgglobal/test/domain/EmailLog.kt
package com.cfgglobal.test.domain

import com.cfgglobal.test.enums.TaskStatus
import com.fasterxml.jackson.annotation.JsonIgnore
import org.hibernate.annotations.DynamicInsert
import org.hibernate.annotations.DynamicUpdate
import java.io.Serializable
import javax.persistence.*


@Entity
@DynamicUpdate
@DynamicInsert

data class EmailLog(

        var subject: String? = null,
        @Lob
        @Column(length = 100000)
        @JsonIgnore
        var content: ByteArray? = null,
        var sendTo: String? = null,
        @Enumerated(value = EnumType.STRING)
        var status: TaskStatus? = null,
        var times: Int? = null,
        var msg: String? = null
) : BaseEntity(), Serializable
